src/plugins/intel_cpu/src/nodes/kernels/kernel_utils.hpp

```diff
@@ -0,0 +1,547 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cpu/x64/jit_generator.hpp"
+#include <ie_common.h>
+#include <dnnl_types.h>
+#include <set>
+
+namespace ov {
+namespace intel_cpu {
+
+template <typename Vmm>
+class vRefWrap;
+
+#define r64Ref() rRefWrap<Xbyak::Reg64>(this, r64Pool[getRegIdx()])
+
+class jitKernelBase: public dnnl::impl::cpu::x64::jit_generator {
```

Comment:
I think this class should be renamed to "snake" case like it is done for emmiters.
Also I have a similar PR that also make it as templates to reduce a boilerplate code https://github.com/openvinotoolkit/openvino/pull/12991/files#diff-92bae914fbb339c50e298ca5899a11d329552b2e39138804f1f22958be991069R20

Reply:
I left one concern there. After that we can merge solutions smoothly.